File: src/main/java/me/nobaboy/nobaaddons/features/chatcommands/impl/dm/PartyMeCommand.java
package me.nobaboy.nobaaddons.features.chatcommands.impl.dm;

import me.nobaboy.nobaaddons.NobaAddons;
import me.nobaboy.nobaaddons.features.chatcommands.ChatContext;
import me.nobaboy.nobaaddons.features.chatcommands.IChatCommand;
import me.nobaboy.nobaaddons.util.ChatUtils;

import java.util.Collections;
import java.util.List;

public class PartyMeCommand implements IChatCommand {
    @Override
    public String name() {
        return "partyme";
    }

    @Override
    public void run(ChatContext ctx) {
        ChatUtils.delayedSend("p " + ctx.user());
    }

    @Override
    public boolean isEnabled() {
        return NobaAddons.config.dmPartyMeCommand;
    }

    @Override
    public List<String> aliases() {
        return Collections.singletonList("pme");
    }
}

File: src/main/kotlin/me/nobaboy/nobaaddons/features/chatcommands/impl/dm/PartyMeCommand.kt
package me.nobaboy.nobaaddons.features.chatcommands.impl.dm

import me.nobaboy.nobaaddons.NobaAddons
import me.nobaboy.nobaaddons.features.chatcommands.ChatContext
import me.nobaboy.nobaaddons.features.chatcommands.IChatCommand
import me.nobaboy.nobaaddons.util.HypixelCommands

class PartyMeCommand : IChatCommand {
    override val name: String = "partyme"

    override val aliases: MutableList<String> = mutableListOf("pme")

    override val isEnabled: Boolean
        get() = NobaAddons.config.dmPartyMeCommand

    override fun run(ctx: ChatContext) {
        HypixelCommands.partyInvite(ctx.user())
    }
}